### PR TITLE
Get rid of gh-pages branch

### PR DESCRIPTION
**Pull request by DRSchlaubi**
Since it works for the wiki, let's try this again

### CODE REVIEW

**Review by Lukellmann — CHANGES_REQUESTED** [review]

File: .github/workflows/docs-ci.yml [hunk 1]
```diff
@@ -1,39 +1,55 @@
...
+        with:
+          path: docs-cache
+          key: docs-cache
       - name: Build docs with Gradle
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: --stacktrace --info dokkaHtmlMultiModule
-      - name: Deploy docs to GitHub Pages
-        uses: JamesIves/github-pages-deploy-action@v4
+          arguments: :produceDocs
```
Comment (Lukellmann) [inline .github/workflows/docs-ci.yml:46]: where does this task come from?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -98,6 +99,7 @@ pluginsForBuildSrc = [
     "kotlin-jvm-plugin",
     "kotlin-serialization-plugin",
     "dokka-plugin",
+    "dokka-versioning-plugin",
```
Comment (Lukellmann) [inline gradle/libs.versions.toml:102]:
```suggestion
```
should not be part of this bundle, it's a library

File: .github/workflows/docs-ci.yml [hunk 3]
```diff
@@ -1,39 +1,55 @@
...
+  pages: write
 
+# This workflow updates the versioned docs on dokka.kord.dev using the following strategy
+# 1) pull cache to docs-cache
+# 2) Call dokkaHtmlMultiModule for current version
+#   - if this is a release build move current docs into docs-cache/$version
+#   - otherwise move it into docs-cache/current
+# 3) Build GitHub Pages artifact
+#   - Move all folders not named "current" into docs/<folder>
+#   - Move current into docs/
```
Comment (Lukellmann) [inline .github/workflows/docs-ci.yml:21]: i don't get where this is done

**Review by Lukellmann — CHANGES_REQUESTED** [review]
the versioning plugin seems to be quite complex to configure (see [here](https://github.com/Kotlin/dokka/tree/master/plugins/versioning#usage-example)), should we maybe separate this into two prs? (one getting rid of the gh-pages branch, another one configuring the versioning plugin)

File: .github/workflows/docs-ci.yml [hunk 4]
```diff
@@ -1,39 +1,55 @@
...
+name: Build docs
 
-name: Docs
-
-on: push # but only to default branch, see if below
+on:
+  push:
+    branches: [ "0.9.x" ]
+  tag:
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (DRSchlaubi) [reply .github/workflows/docs-ci.yml:7]: Yes to trigger the build for the current release

File: .github/workflows/docs-ci.yml [hunk 5]
```diff
@@ -1,39 +1,55 @@
-# This workflow will build and deploy docs to GitHub Pages whenever something is pushed to the default branch
+name: Build docs
 
-name: Docs
-
-on: push # but only to default branch, see if below
+on:
+  push:
+    branches: [ "0.9.x" ]
+  tag:
```
[the remark this answers is not in the document]
Comment (DRSchlaubi) [reply .github/workflows/docs-ci.yml:6]: yes `tag` doesn't exist anyways

File: .github/workflows/docs-ci.yml (hunk 3 above)
Comment (DRSchlaubi) [reply .github/workflows/docs-ci.yml:21]: the produceDocs tasks

File: .github/workflows/docs-ci.yml (hunk 1 above)
Comment (DRSchlaubi) [reply .github/workflows/docs-ci.yml:47]: it's called `:produceDocs` the `:` means it's from the root project, so it's in the root `build.gradle.kts`

File: buildSrc/src/main/kotlin/Documentation.kt [hunk 6]
```diff
@@ -34,4 +36,16 @@ fun AbstractDokkaLeafTask.applyKordDokkaOptions() {
             suppress = true
         }
     }
+    project.dependencies {
+        "dokkaHtmlPartialPlugin"(project.libs.findLibrary("dokka-versioning-plugin").get())
+    }
+    applyVersioningPlugin()
+}
+
+fun AbstractDokkaTask.applyVersioningPlugin() {
```
[the remark this answers is not in the document]
Comment (DRSchlaubi) [reply buildSrc/src/main/kotlin/Documentation.kt:45]: We need to apply this configuration for the root project and the sub projects so putting it just in the root project makes no sense

File: .github/workflows/docs-ci.yml (hunk 3 above)
Comment (Lukellmann) [reply .github/workflows/docs-ci.yml:21]: is it custom? cause if so, you didn't commit it

File: .github/workflows/docs-ci.yml (hunk 1 above)
Comment (Lukellmann) [reply .github/workflows/docs-ci.yml:47]: did you commit it?

File: .github/workflows/docs-ci.yml [hunk 7]
```diff
@@ -1,39 +1,56 @@
...
+        uses: actions/checkout@v3
       - name: Set up JDK
         uses: actions/setup-java@v3
         with:
           distribution: temurin
           java-version: 8
+      - uses: actions/cache@v3
+        with:
+          path: docs-cache
+          key: docs-cache
```
Comment (Lukellmann) [inline .github/workflows/docs-ci.yml:43]: how long will this cache be persisted? because if it's cleared, we no longer have the docs of older versions

Comment (DRSchlaubi) [reply .github/workflows/docs-ci.yml:43]: After 7 days of not being accessed, we might need to change this to something else

**Comment by DRSchlaubi** [comment]
Superseded by #855

**Pull request closed without merging**